lms/djangoapps/teams/views.py

```diff
@@ -570,3 +576,324 @@ def get(self, request, topic_id, course_id):
 
         serializer = TopicSerializer(topics[0])
         return Response(serializer.data)
+
+
+class MembershipListView(ExpandableFieldViewMixin, GenericAPIView):
+    """
+        **Use Cases**
+
+            List course team memberships or add a user to a course team.
+
+        **Example Requests**:
+
+            GET /api/team/v0/team_membership
+
+            POST /api/team/v0/team_membership
+
+        **Query Parameters for GET**
+
+            At least one of username and team_id must be provided.
+
+            * username: Returns membership records only for the specified user.
+              If the requesting user is not staff then only memberships for
+              teams associated with courses in which the requesting user is
+              enrolled are returned.
+
+            * team_id: Returns only membership records associated with the
+              specified team.
```

Comment:
Add doc that requesting user must be staff or in the course containing the team.